src/util/base58.rs

```diff
@@ -65,6 +65,49 @@ impl error::Error for Error {
     }
 }
 
+/// Vector-like object that holds the first 100 elements on the stack. If more space is needed it
+/// will be allocated on the heap.
+struct SmallVec<T> {
+    len: usize,
+    stack: [T; 100],
+    heap: Vec<T>,
+}
+
+impl<T: Default + Copy> SmallVec<T> {
+    pub fn new() -> SmallVec<T> {
+        SmallVec {
+            len: 0,
+            stack: [T::default(); 100],
+            heap: Vec::new(),
+        }
+    }
+
+    pub fn push(&mut self, val: T) {
+        if self.len < 100 {
+            self.stack[self.len] = val;
+            self.len += 1;
+        } else {
+            self.heap.push(val);
+        }
+    }
+
+    pub fn iter<'f>(&'f self) -> Box<DoubleEndedIterator<Item=&'f T> + 'f> {
```

Comment:
I think I couldn't see the forest for the trees here :man_facepalming: of course `Box`es do allocations: the one thing we wanted to reduce with this PR.